### DISCUSSION TITLE
Login problem with Safari prevents publishing on quartopub

### DISCUSSION THREAD

**Discussion by BernardForgues** [quarto-dev/quarto-cli#3994] [Q&A]
Hi all,
For some reason, I cannot sign in to Quarto Pub on Safari (16.2 on macOS Monterey 12.6.2). I can connect on Chrome and Firefox, so the issue is not with my Quarto Pub account.
Now the problem is that the 'quarto publish' command apparently wants to sign in using the default browser (in my case Safari) and fails even though I am logged in on another browser.
Any idea how to solve this (besides ditching Safari

**Comment by cscheid** [comment]
(As a workaround) After the window is open, can you copy and paste the address to your preferred browser?

**Reply by BernardForgues** [reply]
Thanks for answering Carlos!
So, when entering 'quarto publish quarto-pub' in the Terminal, and 'Yes' when prompted to Authorize, RStudio launches a Safari window with this URL: https://quartopub.com/sign-in?source=authorize&id=not_sure_I_should_be_posting_the_actual_id
But when I copy-paste it into Chrome, it loops back to quartopub's home page…

**Reply by cscheid** [reply]
@softwarenerd What do you think?

**Reply by softwarenerd** [reply]
I believe there is an issue with our auth provider and that we will need to make updates to Quarto Pub to resolve this issue. For the moment, the best workaround is:

> (As a workaround) After the window is open, can you copy and paste the address to your preferred browser?

**Comment by JasonPekos** [comment]
can confirm that I'm also having this issue. I think the problem is with the quarto-pub site.

**Comment by monicathieu** [comment]
Writing in to say I am also having this issue with quartopub.com on Safari!

versions: Safari 16.5, OS Monterey 12.5

I can sometimes get an error message about "cannot access the browser sessionStorage" when I attempt to log into quartopub.com using my GitHub account, which leads me to believe Safari cookie handling is fighting with quarto-pub. I still get the problem even when I turn off content blockers for the site, so I think the cookie issue is at least a little deeper than level 1.

is there somewhere else we ought to be posting this to tag the folks in charge of quartopub.com?

**Reply by BernardForgues** [reply]
Sorry to hear that, Monica.
As for me, this was the opportunity to shift from Safari to Firefox as my default browser. This was overdue anyway, I just needed time to sort bookmarks, which I'll do during vacations…
Good luck!

**Reply by fthielen** [reply]
I am wondering the same @monicathieu. I solved this by temporarily switching my default browser to Chrome just for the log-in procedure. Permanently shifting to either Chrome or Firefox only because quarto cannot handle Safari is no option for me.

**Comment by BernardForgues** [comment]
Thank you Brian,
Actually, I eventually decided to shift to Firefox entirely…
Cheers,
Bernard

On Jan 26, 2024, at 23:17, Brian Lambert ***@***.***> wrote:



I believe there is an issue with our auth provider and that we will need to make updates to Quarto Pub to resolve this issue. For the moment, the best workaround is:

(As a workaround) After the window is open, can you copy and paste the address to your preferred browser?

—
Reply to this email directly, view it on GitHub<https://github.com/quarto-dev/quarto-cli/discussions/3994#discussioncomment-8262070>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AI6NDUHWIZ4T3JVJXOD3UTDYQQTOVAVCNFSM6AAAAAAT5442E2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DENRSGA3TA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

Ce message électronique et tous les fichiers attachés qu'il contient sont confidentiels et destinés exclusivement à l'usage de la personne à laquelle ils sont adressés. Si vous avez reçu ce message par erreur, merci de le retourner à son émetteur. Les idées et opinions présentées dans ce message sont celles de son auteur, et ne représentent pas nécessairement celles de l’institution ou les entités affiliées dont dépend l’auteur. La publication, l'usage, la distribution, l'impression ou la copie non autorisée de ce message et des attachements qu'il contient sont strictement interdits ». “This email and any files transmitted with it are confidential and intended solely for the use of the individual or entity to whom they are addressed. If you have received this email in error please return it to the sender. The ideas and views expressed in this email are solely those of its author, and do not necessarily represent the views of the institution or company the author is dependent from. Unauthorized publication, use, distribution, printing or copying of this e-mail or any attached files is strictly forbidden.
